service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -89,73 +88,208 @@ public void setRestTemplate(RestTemplate restTemplate) {
+
+                HttpStatus status=null;
+                if (processedJson.has(META)) {
+                    status = HttpStatus.OK;
+                    return new ResponseEntity<>(processedJson, status);
+                } else if (processedJson.has(JSON_STATUS_CODE)) {
+                    String statusValue = processedJson.get(JSON_STATUS_CODE).toString();
+                    status = HttpStatus.resolve(Integer.parseInt(statusValue));
+                    return new ResponseEntity<>(processedJson, status);
                 } else {
-                    return new ResponseEntity<>(parsedResponse, HttpStatus.BAD_REQUEST);
+                    return new ResponseEntity<>(processedJson, HttpStatus.SERVICE_UNAVAILABLE);
                 }
+
             } else {
-                return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_SERVICE_ERROR),
-                        HttpStatus.SERVICE_UNAVAILABLE);
-            }
-        } catch (REMGenerateException e1) {
-            if (e1.getMessage().contains(Integer.toString(HttpStatus.NOT_ACCEPTABLE.value()))) {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.NOT_ACCEPTABLE);
-            }
-            else if (e1.getMessage().contains(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.EXPECTATION_FAILED);
-            }
-            else if (e1.getMessage().contains(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.EXPECTATION_FAILED);
-            }
-            else if (e1.getMessage()
-                       .contains(Integer.toString(HttpStatus.SERVICE_UNAVAILABLE.value()))) {
-                return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_ER),
-                        HttpStatus.SERVICE_UNAVAILABLE);
+                return createResponseEntity(HttpStatus.BAD_REQUEST,
+                        "Invalid JSON format,expected either single template or array of templates",
+                        JSON_ERROR_STATUS, errorResponse);
             }
-            else {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.UNPROCESSABLE_ENTITY);
+        } catch (REMGenerateException | JsonSyntaxException e) {
+            return handleException(e);
+        }
+    }
+
+
+    /**
+     * To display response in browser or application
+     * @param status          response code for the HTTP request
+     * @param responseMessage the message according to response
+     * @param resultMessage   whatever the result this message gives you idea about that
+     * @param errorResponse   is to collect all the responses here.
+     * @return ResponseEntity
+     */
+    public ResponseEntity<JsonObject> createResponseEntity(HttpStatus status, String responseMessage, String resultMessage,
```

Comment:
The method should be also available in form createResponseEntity(HttpStatus, String, String) because errorResponse is usually passed as an empty object. Thus, it can be created internally.

Reply:
Please, dont' forget about this one.

Reply:
Done